src/components/WebcamCapture/WebcamCapture.js

```diff
@@ -1,41 +1,100 @@
-import React from 'react';
+import React, {Component} from 'react';
 import './WebcamCapture.css';
 import Webcam from 'react-webcam';
 import Notification from './../Notification/Notification';
 import PropTypes from 'prop-types';
+import Model from './../../utils/model';
 
-const WebcamCapture = props => {
-	const height = 400;
-	const width = 400;
+const ML_THRESHOLD = 0.06;
+const height = 400;
+const width = 400;
 
-	const videoConstraints = {
-		width,
-		height,
-		facingMode: 'user'
+const videoConstraints = {
+	width,
+	height,
+	facingMode: 'user'
+};
+
+class WebcamCapture extends Component {
+	state = {
+		isDetecting: true,
+		cameraConnected: false
 	};
 
-	if (props.cameraConnected) {
+	constructor(props) {
+		super(props);
+		this.webcam = React.createRef();
+		this.model = new Model();
+		this.model.load();
+	}
+
+	componentDidMount() {
+		if (navigator.mediaDevices) {
```

Comment:
There's quite a lot going on here, we could break this down into separate methods to make this a bit easier to read

We could also invert the condition to return early, that way we're not increasing the nesting (cyclomatic complexity)

e.g.

```js
componentDidMount() {
  if (!navigator.mediaDevices) {
    console.error('...'); // Ideally state etc, but we're guaranteed this will work on the tablet, it's more for local development
    return;
  }

  this.setupWebcam(); // Ideally a better name, it's doing a few things
  }
}

setupWebcam() {
  navigator.mediaDevices.getUserMedia({ video: true })
    .then(() => {
      this.setState({
        cameraConnected: true,
        isDetecting: false,
      })

      this.setupScreenshotInterval();
    }).catch(() => {
      this.setState({
        isDetecting: false,
      })
    })
}

setupScreenshotInterval() {
  this.screenshotInterval = setInterval(() => {
    // ...
  })
}
```